oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -304,12 +304,15 @@ public String getAccount() {
    */
   @Override
   public byte[] sign(byte[] toSign) {
-    return IamUtils.sign(
-        getAccount(),
-        this,
-        transportFactory.create(),
-        toSign,
-        Collections.<String, Object>emptyMap());
+    try {
+      String account = getAccount();
+      return IamUtils.sign(
+          account, this, transportFactory.create(), toSign, Collections.<String, Object>emptyMap());
+    } catch (SigningException ex) {
+      throw ex;
+    } catch (RuntimeException ex) {
```

Comment:
Let's add a TODO to clean this up that mentions #314

Reply:
done. Also added some JavaDoc to getAccount.